src/primitives/transaction.h

```diff
@@ -562,34 +721,51 @@ class CTransaction
             fOverwintered &&
             nVersionGroupId == OVERWINTER_VERSION_GROUP_ID &&
             nVersion == OVERWINTER_TX_VERSION;
+
         bool isSaplingV4 =
             fOverwintered &&
             nVersionGroupId == SAPLING_VERSION_GROUP_ID &&
             nVersion == SAPLING_TX_VERSION;
-        if (fOverwintered && !(isOverwinterV3 || isSaplingV4)) {
+
+        // It is not possible to make the transaction's serialized form vary on
+        // a per-enabled-feature basis.The approach here is that all
+        // serialization rules for not-yet-released features must be
+        // non-conflicting and transaction version/group must be set to
+        // ZFUTURE_TX_(VERSION/GROUP_ID)
```

Comment:
We **_need_** to change the transaction version because we're adding fields to the transaction, so we should be able to distinguish between transactions that have `vtzein`/`vtzeout` vectors and transactions which don't.

Reply:
The distinction only matters for the purpose of serialization; the extra memory slots for the vectors don't make a difference, right? So the idea is that transaction version should be used only as a parsing/serialization concern, and everywhere else we just use feature flags. The in-memory transaction data structure can support all available features.

Reply:
Right, during deserialization we want to know if we should expect to see TZE-related fields in the transaction and that's why we need to change the transaction version. In all of the logic outside of serialization we can just do things like:

1. If feature is enabled, vtzein/vtzeout vectors can be non-empty.
2. Otherwise they have to be empty

etc.

Reply:
What I mean here is that the transaction serialized form depends upon the transaction version, not upon the enabled feature flags. So instead of having a bunch of distinct future versions for different features, we have a single version that *all* the parsing/serialization changes for *all* of future features must go under. Then, when we decide to actually include a feature, we move its parsing from under the "future" version to under the actual transaction version check.

The upshot of this is that if you have a test network that's only enabling a single one of the ZFUTURE features, the wire format on that network will still include the changes for *all* ZFUTURE features. Is that a problem?